src/Common/src/Interop/OSX/System.Security.Cryptography.Native.Apple/Interop.Keychain.cs

```diff
@@ -0,0 +1,218 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Runtime.InteropServices;
+using System.Security.Cryptography.Apple;
+
+internal static partial class Interop
+{
+    internal static partial class AppleCrypto
+    {
+        [DllImport(Libraries.AppleCryptoNative)]
+        private static extern int AppleCryptoNative_SecKeychainItemCopyKeychain(
+            IntPtr item,
+            out SafeKeychainHandle keychain);
+        
+        [DllImport(Libraries.AppleCryptoNative, EntryPoint = "AppleCryptoNative_SecKeychainCreate")]
+        private static extern int AppleCryptoNative_SecKeychainCreateTemporary(
+            string path,
+            int utf8PassphraseLength,
+            byte[] utf8Passphrase,
+            out SafeTemporaryKeychainHandle keychain);
+
+        [DllImport(Libraries.AppleCryptoNative)]
+        private static extern int AppleCryptoNative_SecKeychainDelete(IntPtr keychain);
+
+        internal static SafeKeychainHandle SecKeychainItemCopyKeychain(SafeKeychainItemHandle item)
+        {
+            var handle = SecKeychainItemCopyKeychain(item.DangerousGetHandle());
+            GC.KeepAlive(item);
+            return handle;
+        }
+
+        internal static SafeKeychainHandle SecKeychainItemCopyKeychain(IntPtr item)
+        {
+            SafeKeychainHandle keychain;
+            int osStatus = AppleCryptoNative_SecKeychainItemCopyKeychain(item, out keychain);
+
+            // A whole lot of NULL is expected from this.
+            // Any key or cert which isn't keychain-backed, and this is the primary way we'd find that out.
+            if (keychain.IsInvalid)
+            {
+                GC.SuppressFinalize(keychain);
```

Comment:
Optimization.  Just removes it from the finalizer pool if it leaks.  Normally an IsInvalid SafeHandle is a candidate for Finalization, then in the Finalizer it determines that since it IsInvalid there's no remaining cleanup work.

Since I know that this type doesn't support handle value mutation, it's safe to do.